text/src/autogluon/text/automm/data/infer_types.py

```diff
@@ -0,0 +1,301 @@
+import collections
+import pandas as pd
+import warnings
+from typing import Union, Optional, List, Dict, Tuple
+from pandas import DataFrame, Series
+from ..constants import NULL, CATEGORICAL, NUMERICAL, TEXT, IMAGE_PATH, MULTICLASS, BINARY, REGRESSION
+
+
+def is_categorical_column(data: pd.Series,
+                          valid_data: pd.Series,
+                          threshold: int = None,
+                          ratio: Optional[float] = None,
+                          oov_ratio_threshold: Optional[float] = None,
+                          is_label: bool = False,
+                          ) -> bool:
+    """Check whether the column is a categorical column.
+
+    If the number of unique elements in the column is smaller than
+
+        min(#Total Sample * ratio, threshold),
+
+    it will be treated as a categorical column.
+
+    Parameters
+    ----------
+    data
+        The column data
+    valid_data
+        Additional validation data
+    threshold
+        The threshold for detecting categorical column
+    ratio
+        The ratio detecting categorical column
+    oov_ratio_threshold
+        The out-of-vocabulary ratio between training and validation.
+        This is used to determine if the column is a categorical column.
+        Usually, a categorical column can tolerate a small OOV ratio
+    is_label
+        Whether the column is a label column.
+
+    Returns
+    -------
+    is_categorical
+        Whether the column is a categorical column
+    """
+    if data.dtype.name == 'category':
+        return True
+    else:
+        if threshold is None:
+            if is_label:
+                threshold = 100
+                oov_ratio_threshold = 0
+                ratio = 0.1
```

Comment:
what if these are provided? Do we want to override the ones passed as the method args?

Reply:
This is a legacy function from previous TextPredictor. We can replace it after aligning our API design between `automm/data/infer_types.py` and `autogluon.common.features.infer_types`.